packages/core/docs/advanced/caching.md

```diff
@@ -1,99 +1,133 @@
-        return exampleCache.getCache()
-      }
+If you follow the above steps and run the application, you won't see any performance difference. This is because only pages with tags are cached.
 
-      const content = await render();
-      exampleCache.storeCache(getTags(), content)
-      return content;
-    },
-    invalidate: ({ req, res }) => {
-      invalidators.forEach(invalidator => invalidator(req))
-    }
-  };
-};
-```
+Please refer to the [Using tags](#using-tags) section for more information.
 
-## Core mechanism
+## How it works?
 
+When the page is requested, the cache driver checks if there is an already rendered page in the cache matching the current route. If there is, it will serve the cached version. Otherwise:
 
-```ts
-import { useCache } from '@vue-storefront/cache';
-import redisDriver from '@vue-storefront/cache-redis'
+1. Current page will be rendered on the server.
+2. If the page contains tags, the result will be saved in the cache.
+3. Page is served to the user.
 
-const cache = useCache(); // Inside setup
-cache.addTags([{ prefix: 'P', value: '123' }]) // adds tags
-cache.setTags((currentTags) => [...currentTags, { prefix: 'P', value: '123' }]) // adds tags, also you can skip previous one
-cache.getTags(); // returns current tags
-```
+## Using tags
 
-## Nuxt compatibility
-Add it to **modules** not **buildModules** - it is very important.
+Tags should be registered in Vue components or composables. During Server Side Rendering, tags registered in the current route are associated with the rendered page.
 
-```js
-['@vue-storefront/cache/nuxt', {
-  server: {
-    invalidateEndpoint: '/cache-invalidate',
-    invalidators: ['./invalidator'],
-    driver: ['./exampleCache', { test: 1 }]
+To add tags, use `useCache` composable from `@vue-storefront/cache` package.
+
+```javascript
+// pages/Category.vue
+import { useCache, CacheTagPrefix } from '@vue-storefront/cache';
+
+export default {
+  setup() {
+    const { addTags } = useCache();
+
+    addTags([
+      { prefix: CacheTagPrefix.View, value: 'category' },
+      { prefix: CacheTagPrefix.Category, value: id },
+      // or
+      { prefix: 'V', value: 'category' },
+      { prefix: 'C', value: id },
+    ]);
   }
-}]
+};
 ```
 
-## Tagging from integration composables
+Tags should be added for all dynamic parts of the page. For example on the category page, tags should be added for:
+  * current category,
+  * all visible subcategories,
+  * all visible products.
 
-```ts
-const factoryParams = {
-  getCacheTags: (response): CacheTags[] => {
-    return [
-      prefix: 'P', value: '123' },
-      prefix: 'P', value: '456' },
-      prefix: 'P', value: '789' }
-    ]
-  }
+**But why?**
+
+When at least one tag associated with the given page is [invalidated](#invalidating-cache), the whole page is removed from the cache. For example, if one of the products is modified or disabled, we should invalidate cache for pages where it's visible:
```

Comment:
some diagram about the invalidation process from draw.io will be super helpful here ;)

Reply:
I'm not sure how it would look like.